rs/replicated_state/src/metadata_state/subnet_call_context_manager.rs

```diff
@@ -781,13 +781,15 @@ impl TryFrom<pb_metadata::EcdsaArguments> for EcdsaArguments {
 pub struct SchnorrArguments {
     pub key_id: SchnorrKeyId,
     pub message: Arc<Vec<u8>>,
+    pub taproot_tree_root: Option<Vec<u8>>,
```

Comment:
Haven't looked at the rest of the PR (only at files under `rs/protobuf` and `rs/replicated_state`), but would it make sense to also put `taproot_tree_root` behind an `Arc`?

Reply:
Possibly? My understanding of the `Arc` on the message is because the message is of unbounded size [*] and this caused problems with memory pressure. Whereas the taproot input is either `None` or if `Some` then the vec is either empty or exactly 32 bytes.

[*] Modulo size restrictions on xnet calls.

Reply:
If we're cloning this any meaningful number of times; and it's not supposed to be mutable; then I'd put it behind an `Arc`, there's nothing to lose by doing that.

But it's just an idea.

Reply:
Changed this to be an `Arc`